app/javascript/pages/account/account-settings.tsx

```diff
@@ -456,7 +457,8 @@ const AccountSettingsPage = () => {
   const { profile, loading, initialStateLoaded } = React.useContext(UserContext)
 
   if (!profile && !loading && initialStateLoaded) {
-    // TODO: Redirect to React sign in page and show a message that user needs to sign in
+    window.sessionStorage.setItem("redirect", "settings")
```

Comment:
thought: We're using `sessionStorage` across various files, and it's not immediately obvious what arguments for `setItem` mean. We should look into consolidating all routing-related `sessionStorage` usage into a single file, probably in `routeUtil.ts`.